zeebe/exporters/elasticsearch-exporter/src/main/java/io/camunda/zeebe/exporter/ElasticsearchExporter.java

```diff
@@ -245,117 +247,117 @@ private void createIndexTemplates() {
       }
       if (index.timer) {
-        createValueIndexTemplate(ValueType.TIMER);
+        createValueIndexTemplate(ValueType.TIMER, version);
       }
       if (index.messageStartEventSubscription) {
-        createValueIndexTemplate(ValueType.MESSAGE_START_EVENT_SUBSCRIPTION);
+        createValueIndexTemplate(ValueType.MESSAGE_START_EVENT_SUBSCRIPTION, version);
       }
       if (index.processEvent) {
-        createValueIndexTemplate(ValueType.PROCESS_EVENT);
+        createValueIndexTemplate(ValueType.PROCESS_EVENT, version);
       }
       if (index.deploymentDistribution) {
-        createValueIndexTemplate(ValueType.DEPLOYMENT_DISTRIBUTION);
+        createValueIndexTemplate(ValueType.DEPLOYMENT_DISTRIBUTION, version);
       }
       if (index.escalation) {
-        createValueIndexTemplate(ValueType.ESCALATION);
+        createValueIndexTemplate(ValueType.ESCALATION, version);
       }
       if (index.signal) {
-        createValueIndexTemplate(ValueType.SIGNAL);
+        createValueIndexTemplate(ValueType.SIGNAL, version);
       }
       if (index.signalSubscription) {
-        createValueIndexTemplate(ValueType.SIGNAL_SUBSCRIPTION);
+        createValueIndexTemplate(ValueType.SIGNAL_SUBSCRIPTION, version);
       }
       if (index.resourceDeletion) {
-        createValueIndexTemplate(ValueType.RESOURCE_DELETION);
+        createValueIndexTemplate(ValueType.RESOURCE_DELETION, version);
       }
       if (index.commandDistribution) {
-        createValueIndexTemplate(ValueType.COMMAND_DISTRIBUTION);
+        createValueIndexTemplate(ValueType.COMMAND_DISTRIBUTION, version);
       }
       if (index.form) {
-        createValueIndexTemplate(ValueType.FORM);
+        createValueIndexTemplate(ValueType.FORM, version);
       }
       if (index.userTask) {
-        createValueIndexTemplate(ValueType.USER_TASK);
+        createValueIndexTemplate(ValueType.USER_TASK, version);
       }
       if (index.processInstanceMigration) {
-        createValueIndexTemplate(ValueType.PROCESS_INSTANCE_MIGRATION);
+        createValueIndexTemplate(ValueType.PROCESS_INSTANCE_MIGRATION, version);
       }
       if (index.compensationSubscription) {
-        createValueIndexTemplate(ValueType.COMPENSATION_SUBSCRIPTION);
+        createValueIndexTemplate(ValueType.COMPENSATION_SUBSCRIPTION, version);
       }
       if (index.messageCorrelation) {
-        createValueIndexTemplate(ValueType.MESSAGE_CORRELATION);
+        createValueIndexTemplate(ValueType.MESSAGE_CORRELATION, version);
       }
       if (index.user) {
-        createValueIndexTemplate(ValueType.USER);
+        createValueIndexTemplate(ValueType.USER, version);
       }
 
       if (index.authorization) {
-        createValueIndexTemplate(ValueType.AUTHORIZATION);
+        createValueIndexTemplate(ValueType.AUTHORIZATION, version);
       }
     }
 
-    indexTemplatesCreated = true;
+    indexTemplatesCreated.add(version);
```

Comment:
❓ So we always try to create the indices on restart?

Reply:
Yes we always try to create on restart, this mimics the behaviour of the old `indexTemplatesCreated `logic where it was a boolean.